ethers-solc/src/lib.rs

```diff
@@ -237,22 +240,26 @@ impl<Artifacts: ArtifactOutput> Project<Artifacts> {
         }
 
         // If there's a cache set, filter to only re-compile the files which were changed
-        let sources = if self.cached && self.paths.cache.exists() {
-            let cache = SolFilesCache::read(&self.paths.cache)?;
+        let (sources, cached_artifacts) = if self.cached && self.paths.cache.exists() {
+            let mut cache = SolFilesCache::read(&self.paths.cache)?;
             let changed_files = cache.get_changed_or_missing_artifacts_files::<Artifacts>(
                 sources,
                 Some(&self.solc_config),
                 &self.paths.artifacts,
             );
 
+            cache.remove_missing_files();
+            let cached_artifacts = cache
+                .read_artifacts::<Artifacts>(&self.paths.artifacts)
+                .unwrap_or_else(|_| BTreeMap::default());
```

Comment:
This is unlikely to fail since we checked that those files exists, but could still fail due to some race conditions, like the files got wiped after we checked that they exist...

I feel we should propagate the error here because if that would fail, we end up with a successful output that lacks some artifacts.

Reply:
The main failure case (outside of a race) for the fallback was when the artifacts folder is deleted, but the cache metadata still exists. 
I'll change it to propagate the error, and fall back to empty cache if there is no artifacts folder. This way if some of the files have been wiped, it'll return the error. There's still a chance that the folder gets deleted at the same time still and then there'd be missing artifacts.